packages/vscode-docker-registries-extension/src/UnifiedRegistryTreeDataProvider.ts

```diff
@@ -85,31 +85,39 @@ export class UnifiedRegistryTreeDataProvider implements vscode.TreeDataProvider<
         this.onDidChangeTreeDataEmitter.fire(undefined);
     }
 
-    public async connectRegistryProvider(): Promise<void> {
-        const picks: (vscode.QuickPickItem & { provider: RegistryDataProvider<unknown> })[] = [];
+    public async connectRegistryProvider(provider: RegistryDataProvider<unknown> | undefined = undefined): Promise<void> {
         const connectedProviderIds = this.storageMemento.get<string[]>(ConnectedRegistryProvidersKey, []);
 
-        for (const provider of this.providers.values()) {
-            if (connectedProviderIds.includes(provider.id)) {
-                continue;
+        if (!provider) {
+            const picks: (vscode.QuickPickItem & { provider: RegistryDataProvider<unknown> })[] = [];
+
+            for (const currentProvider of this.providers.values()) {
+                if (connectedProviderIds.includes(currentProvider.id)) {
+                    continue;
+                }
+
+                picks.push({
+                    label: currentProvider.label,
+                    description: currentProvider.description,
+                    provider: currentProvider
+                });
             }
 
-            picks.push({
-                label: provider.label,
-                description: provider.description,
-                provider: provider
-            });
-        }
+            const picked = await vscode.window.showQuickPick(picks, { placeHolder: vscode.l10n.t('Select a registry provider to use') });
+            if (!picked) {
+                return;
+            }
 
-        const picked = await vscode.window.showQuickPick(picks, { placeHolder: vscode.l10n.t('Select a registry provider to use') });
+            provider = picked.provider;
+        }
 
-        if (!picked) {
-            return;
+        await provider?.onConnect?.();

```

Comment:
Since `onConnect` can have side effects, I'd execute it within that if block below, i.e. only if a provider is being added, not was already added.

Reply:
good catch!